### DISCUSSION TITLE
`CURVE` object in Montgomery curves

### DISCUSSION THREAD

**Discussion by li0ard** [paulmillr/noble-curves#165] [Q&A] [answered]
Is it possible to get the parameters of the montgomery curve through the `CURVE` property?

**Comment by paulmillr** [comment]
Not currently. Which params do you need and why?

**Reply by li0ard** [reply]
I need all the constants of the curve, because i'm developing a module to create birationally equivalent Weierstrass curves from Edwards and Montgomery curves

**Reply by paulmillr** [reply]
It would make sense to expose them if it was real curve methods.

We have very specific code for x25519 / x448. Generic Montgomery curves can't be constructed using it. Hence, this is a way of discouraging doing anything complex.

**Reply by li0ard** [reply]
You can see formulas [here](https://li0ard.rest/apps/ec_conv). Im just trying to port code from sage to noble

**Reply by paulmillr** [reply, marked as answer]
We only have two montgomery curves for now. Just hardcode the params. We won't have any generic methods for the near future, because there is no point to implement them. For example, there are no specs, no real-world usage, etc.

**Comment by li0ard** [comment]
Also possibly off topic, but how do I take a negative number to degree 2 (over x448  prime field)?

In sage i got:
```py
p = 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffeffffffffffffffffffffffffffffffffffffffffffffffffffffffff
K = GF(p)
a = K(0x01)

print(-a^2)
# 726838724295606890549323807888004534353641360687318060281490199180612328166730772686396383698676545930088884461843637361053498018365438
```
in noble:
```ts
const f = Field(0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffedn)
const a = f.create(0x01)

console.log(f.pow(-a, 2n));
// 1n
```

**Reply by paulmillr** [reply]
I'm not sure what you're doing here. `# 726838...` is number which is just p - 1. It does not print the result.

The number is the same as `f.create(-1n)`.
